### DISCUSSION TITLE
Is Pop OS Getting Fat? (RAM usage)

### DISCUSSION THREAD

**Discussion by cdelv** [pop-os/pop#2910] [General]
Hello everyone,

I hope this message finds you all good.

Over the past five years, I have been using Pop OS as my primary operating system on most of my computers. I absolutely love it. I have observed over time an increase in RAM usage right after boot and during light usage compared to when I initially started using it. Personally, this doesn't pose a problem for me as my computers are equipped with a lot of RAM, given my scientific computing needs. However, I would like to know your thoughts on whether this increase in RAM usage is a cause for concern. Is it normal for the OS to continuously consume more and more RAM over time?

I want to highlight that I regularly wipe my disks and perform fresh installations of Pop OS approximately every 4 months. Therefore, the RAM increase in my case cannot be attributed to an accumulation of unnecessary files or programs. I consistently use the same set of programs, so I strongly believe that this RAM increase is specific to the OS itself. I have not encountered any significant performance issues, except in the App Store. So, I can't tell whether this increase in RAM usage actually improves performance. Additionally, I have not noticed any significant changes in the overall appearance of the OS, so  I don't think that cosmetic alterations are attributed to this RAM increase.

I would greatly appreciate your insights and experiences regarding this matter.

Best regards,

**Comment by leviport** [comment]
How are you monitoring RAM usage? We introduced zram a few months ago, and that seems to have made GNOME System Monitor report that much more RAM is being used than is actually being used, for some reason. Right now the machine I'm currently using reports 7.8gb used if I check `free -h`, but System Monitor says I'm using 16gb.

**Comment by cdelv** [comment]
I usually have htop running continuously. That's how I have been noticing this increase over the years. I would say that over the past 5 years, the RAM usage has gone up by about 800 Mb, but I'm not 100 % sure of the exact amount.

**Reply by leviport** [reply]
Gotcha, htop should be accurate, unlike System Monitor.

**Reply by mmstick** [reply]
You should check back with what processes are consuming the most memory. We use roughly the same amount of RAM as we've always required, with the appcenter and gnome-shell consuming the most.

**Comment by thomas-zimmerman** [comment]
On a somewhat new install, I see a large amount of RSS taken by Pop!_Shop (io.elemenatary.appcenter). This is likely not something to fix as it checked for updates on boot and will be one of the only "large" processes to have run on the system. With `zram` compression active, I have noticed more swap being put to use -- which is good. Cold memory applications (like Pop!_Shop), can be pushed out to swap.

On a default install, using `htop` as our measure, the default memory usage looks very close to the same:

20.04 LTS: 1.32G
![image](https://github.com/pop-os/pop/assets/41023059/f88465fa-0498-48ec-a9c1-87bc8917eae9)

22.04 LTS: 1.15G
![image](https://github.com/pop-os/pop/assets/41023059/26d8b764-282b-4238-a7d8-68fe715b1ef7)

Most of the difference that I'm seeing looks like Pop!_Shop improved over time.
